client/state/data-layer/wpcom-http/utils.js

```diff
@@ -42,11 +43,23 @@ export const getHeaders = action => get( action, 'meta.dataLayer.headers', null
  */
 export const getProgress = action => get( action, 'meta.dataLayer.progress', null );
 
+export const makeParser = ( schema, schemaOptions = {}, transformer = identity ) => {
+	const options = Object.assign( { verbose: true }, schemaOptions );
+	const validator = schemaValidator( schema, options );
+	const filter = schemaValidator.filter( schema );
+
+	return responseData => validator( responseData )
+		? [ true, transformer( filter( responseData ) ) ]
```

Comment:
I think we should run this step outside the return if we want to catch transformer or filter throwing. Or do we want to just let that failure bubble out? This is all likely living inside a promise callback, so a bubble may end up getting swallowed...